-Werror is just causing too much problems

here's the build log:
```
        mkdir -p build
        clang -O3 -Wall -Wextra -Wshadow -Wconversion -Wformat=2 -Winline -Wundef        -Wpointer-arith -Wpedantic -Wstrict-aliasing -Wcast-align -march=native        -mtune=native -O2 -Wall -Werror -fpic -std=gnu99 -shared src/fzf.c -o build/libfzf.so
        src/fzf.c:36:1: error: extra ';' outside of a function [-Werror,-Wextra-semi]
           36 | gen_slice(i16, int16_t);
              | ^
        src/fzf.c:31:31: note: expanded from macro 'gen_slice'
           31 |   gen_simple_slice(name, type);                                                \
              |                               ^
        src/fzf.c:36:24: error: extra ';' outside of a function [-Werror,-Wextra-semi]
           36 | gen_slice(i16, int16_t);
              |                        ^
        src/fzf.c:37:31: error: extra ';' outside of a function [-Werror,-Wextra-semi]
           37 | gen_simple_slice(i32, int32_t);
              |                               ^
        src/fzf.c:38:1: error: extra ';' outside of a function [-Werror,-Wextra-semi]
           38 | gen_slice(str, const char);
              | ^
        src/fzf.c:31:31: note: expanded from macro 'gen_slice'
           31 |   gen_simple_slice(name, type);                                                \
              |                               ^
        src/fzf.c:38:27: error: extra ';' outside of a function [-Werror,-Wextra-semi]
           38 | gen_slice(str, const char);
              |                           ^
        src/fzf.c:225:26: error: implicit conversion loses integer precision: 'size_t' (aka 'unsigned long') to 'uint32_t' (aka 'unsigned int') [-Werror,-Wshorten-64-to-32]
          225 |   pos->data[pos->size] = value;
              |                        ~ ^~~~~
        src/fzf.c:247:28: error: implicit conversion loses integer precision: 'size_t' (aka 'unsigned long') to 'uint32_t' (aka 'unsigned int') [-Werror,-Wshorten-64-to-32]
          247 |     pos->data[pos->size] = k;
              |                          ~ ^
        src/fzf.c:369:34: error: unused parameter 'runes' [-Werror,-Wunused-parameter]
          369 | static bool is_ascii(const char *runes, size_t size) {
              |                                  ^
        src/fzf.c:369:48: error: unused parameter 'size' [-Werror,-Wunused-parameter]
          369 | static bool is_ascii(const char *runes, size_t size) {
              |                                                ^
        src/fzf.c:464:66: error: unused parameter 'slab' [-Werror,-Wunused-parameter]
          464 |                                 fzf_position_t *pos, fzf_slab_t *slab) {
              |                                                                  ^
        src/fzf.c:492:16: error: comparison of integers of different signs: 'int32_t' (aka 'int') and 'const size_t' (aka 'const unsigned long') [-Werror,-Wsign-compare]
          492 |       if (pidx == M) {
              |           ~~~~ ^  ~
        src/fzf.c:739:22: error: comparison of integers of different signs: 'size_t' (aka 'unsigned long') and 'int32_t' (aka 'int') [-Werror,-Wsign-compare]
          739 |       if (i > 0 && j >= f.data[i]) {
              |                    ~ ^  ~~~~~~~~~
        src/fzf.c:742:13: error: comparison of integers of different signs: 'size_t' (aka 'unsigned long') and 'int32_t' (aka 'int') [-Werror,-Wsign-compare]
          742 |       if (j > f.data[i]) {
              |           ~ ^ ~~~~~~~~~
        src/fzf.c:772:69: error: unused parameter 'slab' [-Werror,-Wunused-parameter]
          772 |                                    fzf_position_t *pos, fzf_slab_t *slab) {
              |                                                                     ^
        src/fzf.c:836:64: error: unused parameter 'slab' [-Werror,-Wunused-parameter]
          836 |                               fzf_position_t *pos, fzf_slab_t *slab) {
              |                                                                ^
        src/fzf.c:871:64: error: unused parameter 'slab' [-Werror,-Wunused-parameter]
          871 |                               fzf_position_t *pos, fzf_slab_t *slab) {
              |                                                                ^
        src/fzf.c:908:63: error: unused parameter 'slab' [-Werror,-Wunused-parameter]
          908 |                              fzf_position_t *pos, fzf_slab_t *slab) {
              |                                                               ^
        src/fzf.c:993:65: error: unused parameter 'normalize' [-Werror,-Wunused-parameter]
          993 | fzf_pattern_t *fzf_parse_pattern(fzf_case_types case_mode, bool normalize,
              |                                                                 ^
        18 errors generated.
        make: *** [Makefile:24: build/libfzf.so] Error 1
```

--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,4 @@
-CFLAGS += -Wall -Werror -fpic -std=gnu99
+CFLAGS += -Wall -fpic -std=gnu99
 
 ifeq ($(OS),Windows_NT)
     CC = gcc
@@ -25,6 +25,11 @@ build/$(TARGET): src/fzf.c src/fzf.h
 
 build/test: build/$(TARGET) test/test.c
 	$(CC) -Og -ggdb3 $(CFLAGS) test/test.c -o build/test -I./src -L./build -lfzf -lexaminer
+
+.PHONY:
+debug: src/fzf.c src/fzf.h
+	$(MKD) build
+	$(CC) -Og $(CFLAGS) -Werror -shared src/fzf.c -o build/$(TARGET)
 
 .PHONY: lint format clangdhappy clean test ntest
 lint:

